### PR TITLE
fix type comparisons

### PR DESCRIPTION
**Pull request by bernt-matthias**
linter complained about this over here https://github.com/galaxyproject/planemo/pull/1381

Not sure if `isinstance` or `is` is correct?

### CODE REVIEW

File: planemo/workflow_lint.py [hunk 1]
```diff
@@ -191,12 +191,12 @@ def _lint_best_practices(path: str, lint_context: WorkflowLintContext) -> None:
     """
 
     def check_json_for_untyped_params(j):
-        values = j if type(j) == list else j.values()
+        values = j if isinstance(j, list) else j.values()
```
Comment (mvdbeek) [inline planemo/workflow_lint.py:194]: Can you check this is a dict instance instead ? that should be less narrow. Also the check in line 196 isn't really how you'd do this

Comment (bernt-matthias) [reply planemo/workflow_lint.py:194]: > Can you check this is a dict instance instead ? 

So `is TYPE` everywhere?

> Also the check in line 196 isn't really how you'd do this

Also with `is`?

Comment (mvdbeek) [reply planemo/workflow_lint.py:194]: No, type is type is the most narrow of them all which will not work for any subclasses. `.values()` will only work for objects that satisfy Mapping, while any j that is Iterable will work. So check for the specific instance where you need to call `.values()` first `isinstance(j, Mapping)` and don't worry about the other types that will all be fine.

Comment (bernt-matthias) [inline planemo/workflow_lint.py:194]:
```suggestion
        values = j.values() if isinstance(j, dict) else j
```

**Review by mvdbeek — COMMENTED** [review]
Thank you!

**Review by mvdbeek — APPROVED** [review]

**Pull request merged**